syntax = "proto3";

package dataretrieval;


//Run in the working directory: python -m grpc_tools.protoc -I ./ --python_out=. --pyi_out=. --grpc_python_out=. ./retrieval.proto 

service RetrievalService {
  
  //unary rpc
  rpc GetDickeyFuller(RetrievalRequest) returns (RetrievalDickeyFuller) {}

  //client-unary, server-streaming rpc
  rpc ListDecomposition(RetrievalRequest) returns (stream RetrievalDecomposition) {}

  rpc ListHodrickPrescott(RetrievalRequest) returns (stream RetrievalHodrickPrescott) {}

  rpc ListAutocorrelation(RetrievalRequest) returns (stream RetrievalAutocorrelation) {}

  rpc ListAggregates(RetrievalRequest) returns (stream RetrievalAggregates) {}
}

// Il messaggio di richiesta conterrà la query.
message RetrievalRequest {
  string query = 1;
}

// Il messaggio di risposta conterrà i dati in modo strutturato in base alla tabella interrogata.
message RetrievalDickeyFuller {
  string name = 1;
  string partition = 2;
  float test_statistic = 3;
  float p_value = 4;
  bool is_stationary = 5;
  float crit_value1 = 6;
  float crit_value5 = 7;
  float crit_value10 = 8;
}

message RetrievalDecomposition {
  string name = 1;
  string partition = 2;
  string timestamp = 3;
  string typology = 4;
  float value = 5;
}

message RetrievalHodrickPrescott {
  string name = 1;
  string partition = 2;
  string timestamp = 3;
  float value = 4;
}

message RetrievalAutocorrelation {
  string name = 1;
  string partition = 2;
  float value = 3;
}

message RetrievalAggregates {
  string name = 1;
  string partition = 2;
  string range = 3;
  string typology = 4;
  float value = 5;
  float predicted_value = 6;
}
